<launch>
    <arg name="weights"            default="yolov5l.pt" />
    <arg name="data"               default="data/coco128.yaml" />
    <arg name="frames"             default="camera0 camera1 camera2" />
    <arg name="output_image_topic" default="image_objects" />
    <arg name="result_topic"       default="detected_objects" />
    <arg name="image_height"       default="480" />
    <arg name="image_width"        default="640" />
    <arg name="conf_thres"         default="0.60" />
    <arg name="iou_thres"          default="0.45" />
    <arg name="max_det"            default="1000" />
    <arg name="device"             default="0" />
    <arg name="classes"            default="[0, 1, 2, 3, 5, 7]" />
    <arg name="half"               default="False" />
    <arg name="dnn"                default="False" />
    <arg name="name"               default="yolo" />

    <node name="object_detector_$(arg name)" pkg="object_detector" type="detect_node_multi_topic.py">
        <param name="weights"            type="str"    value="$(arg weights)" />
        <param name="data"               type="str"    value="$(arg data)" />
        <param name="frames"             type="str"    value="$(arg frames)" />
        <param name="output_image_topic" type="str"    value="$(arg output_image_topic)" />
        <param name="result_topic"       type="str"    value="$(arg result_topic)" />
        <param name="image_height"       type="int"    value="$(arg image_height)" />
        <param name="image_width"        type="int"    value="$(arg image_width)" />
        <param name="conf_thres"         type="double" value="$(arg conf_thres)" />
        <param name="iou_thres"          type="double" value="$(arg iou_thres)" />
        <param name="max_det"            type="int"    value="$(arg max_det)" />
        <param name="device"             type="int"    value="$(arg device)" />
        <param name="classes"            type="str"    value="$(arg classes)" />
        <param name="half"               type="bool"   value="$(arg half)" />
        <param name="dnn"                type="bool"   value="$(arg dnn)" />
    </node>
</launch>
